src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -1287,6 +1287,88 @@ public static bool TryInferParamsCollectionIterationType(Binder binder, TypeSymb
             return false;
         }
 
+#nullable enable
+        /// <returns>false if the set does not have a unique signature</returns>
+        internal bool FilterMethodsForUniqueSignature(ArrayBuilder<MethodSymbol> methods)
+        {
+            if (methods.Count < 2)
+            {
+                return true;
```

Comment:
> if a candidate applicable in expanded form

I'm not sure what that means in case of inferring delegate signature - we don't have any arguments - so it seems to me any candidate is "applicable in expanded form" if `IsValidParams` returns true for it. And I've used that helper to make the candidates ambiguous if necessary (`hasParamsAndNonParamsMethods`) and it's also already used to infer delegate signature (`GetMethodGroupOrLambdaDelegateType`).

Reply:
> > if a candidate applicable in expanded form
> 
> I'm not sure what that means in case of inferring delegate signature - we don't have any arguments - so it seems to me any candidate is "applicable in expanded form" if `IsValidParams` returns true for it. And I've used that helper to make the candidates ambiguous if necessary (`hasParamsAndNonParamsMethods`) and it's also already used to infer delegate signature (`GetMethodGroupOrLambdaDelegateType`).

I think I touched on this during our offline conversation last week. We can discuss this more offline. This is somewhat related to https://github.com/dotnet/roslyn/pull/71380#pullrequestreview-1997475152, which was alos mentioned during the conversation. I do not think it is a good idea to call ```IsValidParams``` outside of this helper. This helper should  do all the work.

Reply:
Okay, I've refactored to not call IsValidParams outside OverloadResolution during delegate inference.